### PR TITLE
Add Table String Cell with Sanitize HTML

### PR DESCRIPTION
**Pull request by abdulhdr1**
**What is this feature?**

String cell option to tables, with an option to sanitize and render as HTML

**Why do we need this feature?**

It was something present in Grafana < 7, and missing from newer versions 

**Who is this feature for?**

People who want to render icons and links inside tables 

**Which issue(s) does this PR fix?**:

#24482 

**Special notes for your reviewer:**

The last comment on the issue is from @codeincarnate and mentions a possible problem with react-table's virtualized rendering, I didn't hit anything so maybe there's a bug there but I couldn't catch it on my testing environment, I'd appreciate some help on exposing this in a reproducible manner. 

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: packages/grafana-ui/src/components/Table/StringCell.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+import { TableCellProps } from './types';
+
+export function StringCell(props: TableCellProps): JSX.Element {
+  const { cell, tableStyles, cellProps, field, row } = props;
+
+  const sanitizeHTML = field.config.custom?.cellOptions.sanitizeHTML;
+  const hasLinks = Boolean(getCellLinks(field, row)?.length);
+  const clearButtonStyle = useStyles2(clearLinkButtonStyles);
+  const content = sanitizeHTML ? (
```
Comment (ryantxu) [inline packages/grafana-ui/src/components/Table/StringCell.tsx:19]: 😬 -- lets be VERY VERY VERY careful here...  ideally we can just sanitize everything.  if there is a really really good use case for not-sanitized, we need to follow the same pattern as the text panel.   It requires explicit settings in security config to allow this -- and we still get security reports every month telling us how dangerous it is

Comment (abdulhdr1) [reply packages/grafana-ui/src/components/Table/StringCell.tsx:19]: I think it works differently, it looks like the `TextPanel` always sets the content inside `DangerouslySetHtmlContent`, on the string cell when `sanitizeHTML` is off, the cell value gets rendered as text, as React escapes the input by default, only using  `dangerouslySetInnerHTML` when the value is sanitized.

**Comment by larryck** [comment]
Hi @abdulhdr1, what's the status of this PR? Can't wait to give it a try,Sanitize HTML in tables is heavily used in my use case.

**Comment by abdulhdr1** [comment]
I've been waiting on feedback from the grafana team, @ryantxu can you look at this again?

**Comment by Bujupah** [comment]
If you guys are looking for a way to sanitize html completely, you can use rename by regex transformation and configure it using
**Match  :** `\s*(<[^>]+>|<script.+?<\/script>|<style.+?<\/style>)\s*`
**Replace:** `$2` 

image

**Comment by abdulhdr1** [comment]
Hey, @ryantxu @codeincarnate are there any updates on the need for this feature? 
The other issues also seem to have not been updated in a while, please let me know if you got to a different solution that was safer so that I can close this PR or continue it with your feedback.

**Comment by abdulhdr1** [comment]
@yulipumal Unfortunately I couldn't get the necessary feedback from the team about what needed improvement/fix and the PR kind of just died, I'm not working on it anymore.

**Pull request closed without merging**